File: rn-expo/src/application/GetPopularMoviesUseCase.ts
import { MoviePage } from "../domain/entities";
import { MovieRepository, Result } from "../domain/MovieRepository";

export class GetPopularMoviesUseCase {
  constructor(private readonly repository: MovieRepository) {}

  async execute(page: number): Promise<Result<MoviePage>> {
    return this.repository.getPopularMovies(page);
  }
}

File: kmp-native/shared/src/commonMain/kotlin/com/stackbenchmark/kmpnative/domain/usecase/GetPopularMoviesUseCase.kt
package com.stackbenchmark.kmpnative.domain.usecase

import com.stackbenchmark.kmpnative.domain.entity.MoviePage
import com.stackbenchmark.kmpnative.domain.repository.MovieRepository

class GetPopularMoviesUseCase(private val repository: MovieRepository) {
    suspend operator fun invoke(page: Int): Result<MoviePage> =
        repository.getPopularMovies(page)
}
